hydroflow/examples/kvs_bench/broadcast_receiver_stream.rs

```diff
@@ -0,0 +1,57 @@
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::sync::broadcast::error::TryRecvError;
+use tokio::sync::broadcast::Receiver;
+use tokio_stream::Stream;
+
+#[derive(Debug)]
+pub struct ReceiverStream<T: Clone> {
+    inner: Receiver<T>,
+}
+
+impl<T: Clone> ReceiverStream<T> {
+    pub fn new(recv: Receiver<T>) -> Self {
+        Self { inner: recv }
+    }
+}
+
+impl<T: Clone> Stream for ReceiverStream<T> {
+    type Item = T;
+
+    fn poll_next(mut self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        match self.inner.try_recv() {
+            Ok(x) => Poll::Ready(Some(x)),
+            Err(e) => match e {
+                TryRecvError::Empty => Poll::Pending,
+                TryRecvError::Closed => Poll::Ready(None),
+                TryRecvError::Lagged(_) => panic!(),
+            },
+        }
+    }
+}
+
+impl<T: Clone> AsRef<Receiver<T>> for ReceiverStream<T> {
+    fn as_ref(&self) -> &Receiver<T> {
+        &self.inner
+    }
+}
+
+impl<T: Clone> AsMut<Receiver<T>> for ReceiverStream<T> {
+    fn as_mut(&mut self) -> &mut Receiver<T> {
+        &mut self.inner
+    }
+}
+
+impl<T: Clone> From<Receiver<T>> for ReceiverStream<T> {
+    fn from(recv: Receiver<T>) -> Self {
+        Self::new(recv)
+    }
+}
+
+impl<T: Clone> Clone for ReceiverStream<T> {
+    fn clone(&self) -> Self {
+        Self {
+            inner: self.inner.resubscribe(),
+        }
+    }
+}
```

Comment:
Reason not to use https://docs.rs/tokio-stream/0.1.12/tokio_stream/wrappers/struct.BroadcastStream.html ? I see you have a bit more impls here

Reply:
Likely the only reason was because I didn't know it existed, looks like it should do what I want.

Reply:
Actually I do remember looking for something like this, but I was using my IDE autocomplete to enumerate the possibilities for me, so by typing tokio_stream::wrappers:: to get it to show what is there. BroadcastStream is gated behind the sync feature flag though, so my ide did not show it as an option :)